File: tests/fixtures/dart/basics.dart
class User {
  final int id;
  final String name;
  final String? email;
  
  User(this.id, this.name, this.email);
  
  String greet() {
    return "Hello, $name!";
  }
}

void main() {
  final user = User(1, "Alice", "alice@example.com");
  print(user.greet());
}

File: tests/fixtures/nim/basics.nim
type
  User = object
    id: int
    name: string
    email: Option[string]

proc greet(user: User): string =
  "Hello, " & user.name & "!"

let user = User(id: 1, name: "Alice", email: some("alice@example.com"))
echo greet(user)